lib/db/mysql.js

```diff
@@ -283,8 +283,52 @@ module.exports = function (log, error) {
         deviceInfo.name,
         deviceInfo.type,
         deviceInfo.createdAt,
-        deviceInfo.callbackURL
-      ]
+        deviceInfo.callbackURL,
+        deviceInfo.callbackPublicKey
+      ],
+      null,
+      function (err) {
+        if (err.errno === ER_DUP_ENTRY) {
+          if (err.toString().match(/ for key 'UQ_devices_sessionTokenId'$/)) {
```

Comment:
Well this is embarassing. It turns out that this requirement was entirely manufactured inside my own imagination. I am deleting the weirdness.